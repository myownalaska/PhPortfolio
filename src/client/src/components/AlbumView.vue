<template>
  <div class="album">
    <div class="info-block">
      <h2 class="name">{{ album.name }}</h2>
      <p class="description">{{ album.description }}</p>
      <p class="date">{{ formattedDate }}</p>
      <p class="model-name">{{ modelName }}</p>
    </div>
    <img v-for="photo in 5" :key="photo" class="photo" src="../assets/katya.jpeg">
    <div class="controls">
      <a class="next" href="#">Next</a>
      <p class="name-secondary">{{ album.name }}</p>
      <p class="model-name">{{ modelName }}</p>
      <p class="location">{{ album.location }}</p>
      <p class="socials">Follow Vladislav</p>
    </div>
  </div>
</template>

<script>
import gql from 'graphql-tag'
import { formatDate } from "../utils/formatDate";

export default {
  name: "AlbumView",
  data() {
    return {
      album: { model: {} }
    };
  },
  apollo: {
    album: {
      query: gql`query Album($id: ID!) {
        album(id: $id) {
          name,
          description,
          date,
          location,
          model {
            firstName,
            lastName
          }
        }
      }`,
      variables() {
        return { id: this.$route.params.id };
      }
    },
  },
  computed: {
    formattedDate() {
      return formatDate(this.album.date);
    },
    modelName() {
      const { model } = this.album;
      return `${model.firstName} ${model.lastName}`;
    }
  },
};
</script>

<style scoped>
.album {
  height: 100%;
  box-sizing: border-box;
  display: flex;
  padding: 32px;
  overflow-y: scroll;
}

.info-block {
  display: flex;
  flex-direction: column;
  margin-right: 56px;
}

.name {
  margin: 0;
  font-size: 96px;
}

.description {
  margin: 0;
  color: #b2b1b1;
  font-size: 32px;
}

.date {
  margin: auto 0 0 0;
}

.model-name,
.location {
  margin: 0;
  color: #b2b1b1;
}

.photo {
  max-height: 100%;
  margin-right: 72px;
}

.photo:last-of-type {
  margin-right: 0;
}

.controls {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 72px 216px;
}

.next {
  margin-bottom: 16px;
  color: black;
  font-size: 72px;
  text-decoration: none;
}

.name-secondary {
  margin: 0;
  color: black;
  font-size: 24px;
}

.socials {
  margin: auto 0 0 0;
  color: black;
  font-size: 18px;
}
</style>
