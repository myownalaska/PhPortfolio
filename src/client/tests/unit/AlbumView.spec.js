import { mount } from '@vue/test-utils';
import AlbumView from '@/components/AlbumView.vue';

const FAKE_ALBUM = {
  id: 1,
  name: 'Fake Album',
  description: 'It is not real',
  date: '2017-06-22',
  location: 'ST. PETERSBURG',
  model: {
    firstName: 'Your',
    lastName: 'Name',
  },
};

const $route = {
  params: { id: 1 },
};

describe('AlbumView.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(AlbumView, { mocks: { $route } });
    wrapper.setData({ album: FAKE_ALBUM });
    expect(wrapper.text()).toMatchSnapshot();
  });
});
