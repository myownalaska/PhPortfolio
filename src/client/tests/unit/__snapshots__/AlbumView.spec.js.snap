// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumView.vue renders correctly 1`] = `"Fake Album It is not real 22 June Your Name  Next Fake Album Your Name ST. PETERSBURG Follow Vladislav"`;
